src/test/lib/log-handler-test.js

```diff
@@ -0,0 +1,203 @@
+                    loggerTesting.transports.console;
+                    xdone();
+                });           
+                 done();
+                });
+
+        });
+
+        describe('createGroupInstance.', function () {
+
+                it('Should write to console by default & colorize', function (done) {
+                const loggerTesting = logger.createGroupInstance('log-file');
+               assert.equal(loggerTesting.transports.console.colorize, true);
+                   done();
+                });
+
+                it('Should not write to console if consoleFlage=false', function (done) {
+                const loggerTesting = logger.createGroupInstance('log-file','',false);
+                assert.throws(function (xdone) {
+                    loggerTesting.transports.console;
+                    xdone();
+                    });          
+                    done();
+                });
+
+                it('Should create the proper path if given', function (done) {
+                    const loggerTesting = logger.createGroupInstance('log-file','Sub-test');
+                    assert.ok(fs.lstatSync(path.join(__dirname,"../../../logs/Sub-test")).isDirectory());
+                    //loggerTesting.info('asfwdgw');
+                    done();
+                });
+
+                it('Should have createGroupInstance', function (done) {
+                    assert.ok(logger.createGroupInstance);
+                    done();
+                });
+    
+                it('createInstance should be type function', function (done) {
+                    assert.equal(typeof logger.createGroupInstance, 'function');
+                    done();
+                });
+    
+                it('Should not throw an error if fileName is present', function (done) {
+                    assert.doesNotThrow(function () {
+                        logger.createGroupInstance('log-file');
+                    });
+                    done();
+                });
+    
+                it('Should return a winston logger object', function (done) {
+                    const loggerTesting = logger.createGroupInstance('log-file');
+                    assert.ok(loggerTesting instanceof winston.Logger);
+                    done();
+                });
+
+                it('Should return a winston logger object with all 3 levels', function (done) {
+                    const loggerTesting = logger.createGroupInstance('log-file');
+                    assert.equal(loggerTesting.transports['log-file-info'].level, 'info');
+                    assert.equal(loggerTesting.transports['log-file-error'].level, 'error');
+                    assert.equal(loggerTesting.transports['log-file-warn'].level, 'warn');
+                    done();
+                });  
+                it('Should return error if no parameters are given', function (done) {
+                    assert.throws(function (xdone) {
```

Comment:
is `xdone` required?

Reply:
I think so, since it expects an error to be caught and should wait for it.